store/src/main/java/org/apache/rocketmq/store/queue/ConsumeQueueStore.java

```diff
@@ -341,6 +343,50 @@ private void putConsumeQueue(final String topic, final int queueId, final Consum
         }
     }
 
+    public void truncateTimerConsumerOffset(TimerMessageStore timerMessageStore) {
+        if (this.consumeQueueTable.get(TimerMessageStore.TIMER_TOPIC) == null
+            || this.consumeQueueTable.get(TimerMessageStore.TIMER_TOPIC).get(0) == null) {
+            return;
+        }
+        ConsumeQueueInterface consumerQueue = this.consumeQueueTable.get(TimerMessageStore.TIMER_TOPIC).get(0);
+        if (timerMessageStore.getQueueOffset() > consumerQueue.getMaxOffsetInQueue()) {
+            timerMessageStore.setQueueOffset(consumerQueue.getMaxOffsetInQueue());
+        }
+    }
```

Comment:
It would be better to handle timer consumer offset separately in consume queue store.